src/main/daml/Daml/Finance/Settlement/Batch.daml

```diff
@@ -9,25 +9,22 @@ import DA.List (groupOn)
 import DA.Optional (catOptionals, fromOptional)
 import DA.Set (Set)
 import DA.Set qualified as S (empty, member, fromList, insert, intersection, null, size, union)
-import DA.Traversable qualified as T
+import DA.Traversable qualified as T (mapA)
+import Daml.Control.Monad.Trans.State
 import Daml.Finance.Interface.Holding.Base qualified as Holding (I)
-import Daml.Finance.Interface.Holding.Util (undisclose)
 import Daml.Finance.Interface.Settlement.Batch qualified as Batch (Cancel(..), I, Settle(..), View(..))
-import Daml.Finance.Interface.Settlement.Instruction qualified as Instruction (Cancel(..), Execute(..), I)
-import Daml.Finance.Interface.Settlement.Types (Allocation(..), Approval(..), InstructionKey(..), RoutedStep(..))
+import Daml.Finance.Interface.Settlement.Instruction qualified as Instruction (Cancel(..), Execute(..), I, V)
+import Daml.Finance.Interface.Settlement.Types (Allocation(..), Approval(..), BatchKey(..), InstructionKey(..), RoutedStep(..))
 import Daml.Finance.Interface.Types.Common.Types (Id(..), Parties)
-import Daml.Finance.Settlement.Instruction (Instruction(..), mustAuthorizeHelper)
+import Daml.Finance.Interface.Util.Disclosure (undisclose)
+import Daml.Finance.Interface.Util.SyntheticKey (IndexTable, fetch, exercise, fetchByKey)
+import Daml.Finance.Interface.Util.SyntheticKey qualified as SyntheticKey (I, View(..))
+import Daml.Finance.Settlement.Instruction (mustAuthorizeHelper)
+import Prelude hiding (archive, create, exercise, exerciseByKey, fetch, fetchByKey, fetchFromInterface, lookup, lookupByKey)
 
 -- | Type synonym for `Batch`.
 type T = Batch
 
--- | Type for batch key.
-data BatchKey = BatchKey
```

Comment:
I thought we weren't really using the key for batches, is there a specific reason to move this to an interface package? Is it because `HasSyntheticKey` is only well-defined for interfaces?

Reply:
As the interfaces define functions
```
-- | Convert the account's 'View' to its key.
toKey : V -> BatchKey
toKey v = BatchKey with requestors = v.requestors; id = v.id

-- | HIDE
instance HasSyntheticKey I BatchKey where
  getKey = Daml.Finance.Interface.Settlement.Batch.getKey
  getKeyFromCid viewer cid = toKey <$> exercise cid GetView with viewer
```
where the `BatchKey` is reference, the `BatchKey` would (like all other key data types) need to live in an interface package as well (to not rely on an implementation package).

 But it is true that we haven't really made any use of the `BatchKey`, so we might as well consider ripping it out. What do you think?